container_pipeline/pipeline.py

```diff
@@ -3,60 +3,137 @@
 import sys
 import time
 
-import container_pipeline.utils as utils
 from container_pipeline.lib.log import load_logger
 from container_pipeline.lib.openshift import Openshift, OpenshiftError
 from trigger_dockerfile_lint import trigger_dockerfile_linter
+from container_pipeline.utils import get_project_name, get_job_hash
 
 
-def create_project(appid, jobid, repo_url, repo_branch, repo_build_path,
-                   target_file, notify_email, desired_tag, depends_on,
-                   test_tag):
-    job_name = utils.get_job_name({
-        'appid': appid, 'jobid': jobid, 'desired_tag': desired_tag})
-    project = utils.get_job_hash(job_name)
+def create_new_job():
+    """
+    Creates new job (a dictionary) with default value as `None` for all the
+    keys. This is the central place for creation of a job for container
+    pipeline service
+    """
+    job = dict.fromkeys([
```

Comment:
grouping the similar/related keys would be better.

Reply:
how about making it chronological to make searching/accessing easier? Like this:
```python
job = dict.fromkeys(['action',
             'appid',
             'beanstalk_server',
             'build_status',
             'delivery_status',
             'depends_on',
             'desired_tag',
             'dockerfile',
             'image_name',
             'image_under_test',
             'job_name',
             'jobid',
             'last_run_timestamp',
             'lint_status',
             'logs_URL',
             'logs_dir',
             'logs_file_path',
             'namespace',
             'notify_email',
             'output_image',
             'project_hash_key',
             'project_name',
             'repo_branch',
             'repo_build_path',
             'repo_url',
             'retry',
             'retry_delay',
             'scan_status',
             'target_file',
             'test_tag'])
```